tests/testsuite/publish.rs

```diff
@@ -4103,10 +4103,183 @@ fn multiple_unpublishable_package() {
 
     p.cargo("publish -Zpackage-workspace")
         .masquerade_as_nightly_cargo(&["package-workspace"])
+        .replace_crates_io(registry.index_url())
+        .with_stderr_data(str![[r#"
+[WARNING] nothing to publish, but found 2 unpublishable packages
```

Comment:
This test isn't serving is no longer serving its purpose. We should add a publishable package to it and add a new test for "all are unpublishable".